patches/llvm-project/0007-libunwind-Support-aarch64-without-FPU.patch

```diff
@@ -0,0 +1,190 @@
+From 6507d75387f0d5b7f926d5a60a238e930c9453b3 Mon Sep 17 00:00:00 2001
+From: Keith Packard <keithp@keithp.com>
+Date: Fri, 4 Oct 2024 21:06:37 -0700
+Subject: [PATCH 6/7] [libunwind] Support aarch64 without FPU
+
+ldp and stp instructions both require an FPU. Use pairs of ldr or str
+instructions when the target doesn't have one.
+
+Signed-off-by: Keith Packard <keithp@keithp.com>
+---
+ libunwind/src/UnwindRegistersRestore.S | 36 +++++++------
+ libunwind/src/UnwindRegistersSave.S    | 70 ++++++++++++++------------
+ 2 files changed, 61 insertions(+), 45 deletions(-)
+
+diff --git a/libunwind/src/UnwindRegistersRestore.S b/libunwind/src/UnwindRegistersRestore.S
+index 180a66582f..13a40d080c 100644
+--- a/libunwind/src/UnwindRegistersRestore.S
++++ b/libunwind/src/UnwindRegistersRestore.S
+@@ -633,6 +633,12 @@ Lnovec:
+ .arch_extension gcs
+ #endif
+ 
++#if defined(__ARM_FP) && __ARM_FP != 0
++#define LDP(a,b,r,o,p) stp a, b, [r, o]
++#else
++#define LDP(a,b,r,o,p) ldr a, [r, o] ; ldr b, [r, p]
++#endif
++
+ //
+ // extern "C" void __libunwind_Registers_arm64_jumpto(Registers_arm64 *);
+ //
+@@ -642,23 +648,24 @@ Lnovec:
+   .p2align 2
+ DEFINE_LIBUNWIND_FUNCTION(__libunwind_Registers_arm64_jumpto)
+   // skip restore of x0,x1 for now
+-  ldp    x2, x3,  [x0, #0x010]
+-  ldp    x4, x5,  [x0, #0x020]
+-  ldp    x6, x7,  [x0, #0x030]
+-  ldp    x8, x9,  [x0, #0x040]
+-  ldp    x10,x11, [x0, #0x050]
+-  ldp    x12,x13, [x0, #0x060]
+-  ldp    x14,x15, [x0, #0x070]
++  LDP(x2, x3, x0, #0x010, #0x018)
```

Comment:
The LDP instructions with X registers should be available without the FPU, is this part of the change needed?

Reply:
Nope. I'm not sure why I thought that was necessary. I'll be updating the upstream PR once I've finish testing.